### ISSUE TITLE
Getting forbidden error when trying to join dialog

### ISSUE THREAD

**Issue by habiiev** [QuickBlox/quickblox-react-native-sdk#44]
When trying to call joinDialog on one of the dialogs, that the user is an occupant of, I am getting a `The operation couldn’t be completed. (forbidden error 403.)`.

My steps before trying to join:

Firstly I get the session from the BE and call

```javascript
await QB.auth.setSession(session);
```

After that I call:

```javascript
await QB.chat.connect({ password, userId });
await QB.settings.enableAutoReconnect({ enable: true });
```

After it connects I get an event from chat event emitter that the user is connected to chat servers, at which point I try to join a dialog:

```javascript
await QB.chat.joinDialog({ dialogId: 'some dialog id' });
```

And get the following error:

```
The operation couldn’t be completed. (forbidden error 403.)
```

The user is in the list of occupants of the chat I am trying to connect to, and when I  run `getDialogMessages` everything is fine, and I don't get any errors.

Is there something wrong with what I am doing, or is this supposed to work?

**Comment by ghost** [comment]
Hello @habiiev,
Could you please provide more information?
On which platform this is happening (Android, iOS, both)?
What version of React Native SDK you are using?
What data is in the `session` variable from your example (`await QB.auth.setSession(session);`)?
Are you able to retrieve a list of dialogs using `QB.chat.getDialogs()`?

**Comment by habiiev** [comment]
Hi @oleksandr-shvetsov-qb!
This is happening on iOS, haven't tested it on android yet, but I will be sure to tell you when I get to it
Versions:
```
react-native: 0.64.2
quickblox-react-native-sdk: 0.5.13-beta
```

Here is what I get from calling `await QB.auth.setSession(session);` (I redacted some info)

```
{
    "applicationId": 94...,
    "expirationDate": "2021-11-19T15:44:29Z",
    "token": "eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9....._nzhIM4Hnb6IrR8Y5U6ttrjKiag_vrOdamaw4FfzNueLK3bAAEdW594H-eeSybZ1CZAn17ibH6ZTQ0vVX5H_6g",
    "userId": 13198....
}
```

And yes, when calling `QB.chat.getDialogs()` everything is fine, it gives me the correct dialogs. The dialogs that I tried to connect to are there as well.

After some further investigation that I did, I realized that if I call `QB.settings.init({ ... })` - then everything works as expected. But due to security concerns, our team does not want to provide secret tokens on the client-side. Any way that we could work with quickblox without providing it's secrets to the client?

**Comment by ghost** [comment]
@habiiev 
I wasn't able to reproduce the issue you described.
Could you please provide more details to identify where the problem is? e.g. logs from XCode, information about dialog from admin panel (the dialog you are trying join to)

**Comment by habiiev** [comment]
@oleksandr-shvetsov-qb so I couldn't find any logs or any errors anywhere, so it's probably me doing something wrong.
Here is a basic repo with the same behaviour: https://github.com/habiiev/quickblox-forbidden-err-repro.
It just has a basic login page and it doesn't work correctly :(
I can log in on the server, set the session, connect to chat and then I get the error when I am trying to join a dialogue. I would be super grateful if you could point out what I am doing incorrectly.

**Comment by ghost** [comment]
@habiiev 
Thanks for providing the repository with code sample to reproduce the issue. I managed to reproduce this problem, we will keep you informed about the solution to the problem.

**Comment by habiiev** [comment]
@oleksandr-shvetsov-qb Hi! Any updates on this?

**Comment by ghost** [comment]
Hello @habiiev,
Fix will be included in the next release of SDK planned on January 2022.

**Comment by ghost** [comment]
Hi @habiiev,
We have released new version of React Native SDK ([0.5.17-beta](https://github.com/QuickBlox/quickblox-react-native-sdk/releases/tag/0.5.17-beta)), signature of `QB.auth.setSession()` has changed a bit - no need to pass `expirationDate` any more.
Anyway, to solve the issue you were facing you should add call to `QB.settings.init()` before calling any other methods of QuickBlox React Native SDK. QuickBlox React Native SDK will not work correctly if it wasn't initialized first. So your code should look like following:
```javascript
await QB.settings.init({
  appId: '-----',
  authKey: '------',
  authSecret: '---------',
  accountKey: '---------',
});
await QB.auth.setSession({
  applicationId: data.session.application_id,
  token: data.session.token,
  userId: data.session.user_id,
});
await QB.chat.connect({
  userId: data.session.user_id,
  password: password,
});
```
